go_backend/main.go

```diff
@@ -0,0 +1,242 @@
+		for rows.Next() {
+			err := rows.Scan(&userID)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+		if userID == 0 {
+			c.JSON(200, gin.H{
+				"httpCode": "404",
+				"message":  "Such username or password is incorrect",
+			})
+		} else {
+			c.JSON(200, gin.H{
+				"httpCode": "200",
+				"userID":   userID,
+				"message":  "User Found",
+			})
+		}
+
+	}
+	defer rows.Close()
+}
+
+//resetPassword
+func queryResetPassword(c *gin.Context) {
+	//getting data from request
+	email := c.Query("email")
+	password := c.Query("password")
+	rePassword := c.Query("rePassword")
+
+	//connect to the db
+	db, err := sql.Open("mysql", "root:@tcp(127.0.0.1:3306)/foodapp")
+	if err != nil {
+		fmt.Println("DB connection error")
+		c.JSON(200, gin.H{
+			"httpCode": "500",
+			"message":  "Database is not connected",
+		})
+		panic(err.Error())
+	} else {
+		fmt.Println("DB Connected!")
+	}
+	defer db.Close()
+
+	//update to db
+	if rePassword != password {
+		c.JSON(200, gin.H{
+			"httpCode": "500",
+			"message":  "Password and confirmed password do not match",
+		})
+	} else {
+		update, err := db.Query("UPDATE userinfo SET password = ? WHERE email = ?", password, email)
+		if err != nil {
+			fmt.Println("update error")
+			c.JSON(200, gin.H{
+				"httpCode": "500",
+				"message":  "Update is not vailed",
+			})
+			panic(err.Error())
+		} else {
+			c.JSON(200, gin.H{
+				"httpCode": "200",
+				"message":  "Password is updated",
+			})
+
+		}
+		defer update.Close()
+	}
+}
+
+//donor company sighup function
+func queryCompanySignUp(c *gin.Context) {
+	//getting data from request
+	//userID := c.Query("userID")
+	companyName := c.Query("companyName")
+	address := c.Query("address")
+	city := c.Query("city")
+	state := c.Query("state")
+	zipCode := c.Query("zipCode")
+	fedID := c.Query("fedID")
+	einID := c.Query("einID")
+
+	//connect to the db
+	db, err := sql.Open("mysql", "root:@tcp(127.0.0.1:3306)/foodapp")
+	if err != nil {
+		fmt.Println("DB connection error")
+		c.JSON(200, gin.H{
+			"httpCode": "500",
+			"message":  "Database is not connected",
+		})
+		panic(err.Error())
+	} else {
+		fmt.Println("DB Connected!")
+	}
+	defer db.Close()
+
+	//insert to db
+	insert, err := db.Query("INSERT INTO companyInfo VALUES(DEFAULT,?,?,?,?,?,?,?)", companyName, address, city, state, zipCode, fedID, einID)
+
+	if err != nil {
+		fmt.Println("Sign up error")
+		c.JSON(200, gin.H{
+			"httpCode": "500",
+			"message":  "Signup is not vailed",
+		})
+		panic(err.Error())
+	} else {
+		c.JSON(200, gin.H{
+			"httpCode": "201",
+			"message":  "Company has been assigned or created",
+		})
+	}
+	defer insert.Close()
+}
+
+// func pathParameters(c *gin.Context) {
+// 	name := c.Param("name")
+// 	age := c.Param("age")
+
+// 	c.JSON(200, gin.H{
+// 		"name": name,
+// 		"age":  age,
+// 	})
+// }
+func main() {
+	fmt.Println("Hello World")
+
+	r := gin.Default()
+	r.GET("/", homePage)
+	// r.POST("/", postHomePage)
+	r.GET("/signup", querySignUp)
```

Comment:
Lett's group all login related API into users API. "/signup" => "/api/users/signup"

Reply:
Check this out [Google API Guide](https://cloud.google.com/apis/design), and [Google Gmail sample](https://developers.google.com/gmail/api/reference/rest)